bringing development in line with master

monicathieu opened the pull request:
stuff has been uploaded straight to master since I started working on the project, so now bringing development in line so others can fork from this

Commit:
adding todo: remove ddply calls for speed

Commit:
also replace "(null)"

Commit:
slight changes for transform_pulvinar internals

Commit:
implement STRINGENT criterion for extraneous pid rejection

Commit:
implement explicit detection of study conds (e.g. pre, post) in emailed datasets

Commit:
v1.4.2: stringent pid rejection, mark study conds, etc

Commit:
improve support for specifying by-file study conditions

Commit:
improve get_proc_info implementation

Commit:
removing blank PID patch for time being

Commit:
aesthetics

Commit:
doesn't call clean_invalid_subs (for now)

Commit:
improving parse_subsections_pulvinar (add less stringent pid search)

Commit:
improving speed for large emailed datasets

Commit:
fix param name in docs

josegallegos07 commented:
There's a bunch of [tests failing](https://travis-ci.org/josegallegos07/aceR). You may want to update the [tests](https://github.com/josegallegos07/aceR/tree/master/tests) to account for all the new changes you've added.

monicathieu commented:
Yes definitely--in process of doing so. sorry for any notification spam you may have received!

Commit:
standardize "name"

Commit:
fix sorted files implementation

Commit:
implementing dplyr respellings

Commit:
implementing dplyr respelling; one function handles overall and by condition now

Commit:
brief catch to filter datasets w/ too few trials (not real data)

Commit:
accommodating dplyr respelling in stats-apply

Commit:
accommodating dplyr respelling

Commit:
fixing targ/nontarg status

Commit:
now retains every id's dataset w/ diff timestamps

Commit:
will label modules w/ spaces in file names e.g. spatial span

Commit:
recode col_late_response, accommodate old logfile mislabeling of gender

Commit:
bug fixes, ugly monkey patches

Commit:
bugfixes, adding params to feed forward, ugly monkey patches

Commit:
fixing too-short dataset rejection for filter only

Commit:
fixed bug where some NR trials in logfiles were delimiting data in standardize_session_info

Commit:
changing too few trials to scrub <50% (vs <75%) of max

Commit:
adding support for ishihara, spatial cueing modules

Commit:
to correct variations in raw column spelling

Commit:
now subset multiple pid datasets by time gameplayed

Commit:
lousy patch for new email spatial span data having braces

Commit:
adding param which_modules to specify which modules in a folder to process

Commit:
adding scrub_short flag to proc_by_module

Commit:
updating docs

Commit:
adding sample files for testing

Commit:
updating mans

Commit:
removing dupe testing files (oops)

Commit:
reformatting to fix data.table calls

Commit:
adding imports

Commit:
actually no changes here

Commit:
adding pulvinar test

Commit:
adding data for pulvinar test

Commit:
fixing imports, removing package:: calls

Commit:
adding wide output format to proc_by_module()

Commit:
v1.4.3: speed improvements, added wide output to proc_by_module()

Commit:
add param for user to specify which modules to output

Commit:
removing support for excel file reading in at this time; XLConnect has Java issues

Commit:
adding global var for all modules together

Commit:
fixing implementation of wide vs list output

Commit:
removing irrelevant output from brt, spans, filter

Commit:
throws warning now for excel files (XLConnect problem)

Commit:
adding handedness as a standardized col name

Commit:
putting is_excel() in here cause load-xls is deprecated for now

Commit:
adding scratchpad to ignores

Commit:
adding ACE T1 & T3 key norming dists as internal data

Commit:
calculating boxed "score" in the module

Commit:
moving module-composites into R to officially build functions

Commit:
adding personal *.Rproj to ignores

Commit:
making input data expectations more robust

Commit:
recoding brt by dominant/nondominant now functional!

Commit:
adding ability to scrub within-subject "outlier" RTs. currently only by >2sd from within-subj mean

Commit:
version up to 1.4.4, actually exporting composite functions

Commit:
versioning up officially

Commit:
updating roxygen documentation

Commit:
renaming to accommodate SEA versions

Commit:
adding SEA loading functionality!

Commit:
renaming to accommodate sea version

Commit:
small doc style

Commit:
Versioning up to 2.0.0: loads SEA educational task data! (Soon will process too)

Commit:
unit testing for SEA loading functions

Commit:
fixing roxygen whoopsies

Commit:
not testing load-xls because I know for sure it doesn't work (unsupported package dependencies)

Commit:
doesn't rename "fractions" module in module column

Commit:
version up to 2.0.1; within-subj outlier RT removal now works by SD or by specified range

Commit:
creating new files for sea module proc

Commit:
renaming proc_by_module to proc_ace_by_module

Commit:
now doesn't distinguish by file of origin (when rejecting dupes)

Commit:
adding half, fixing time so not diff on every trial, standardizing module names to underscores

Commit:
creating basically

Commit:
now checks for RW to do RW processing

Commit:
nothing to see here

Commit:
added try() to allow apply_stats with no conditions

Commit:
create ace_sum

Commit:
creating sea_descriptive_statistics (calcs sum too)

Commit:
reorganizing location of math helpers

Commit:
supports reading fluency & reading comprehension

Commit:
no longer turns NAs to "" in transform_grouping_rows (spatial span sometimes bug)

Commit:
fixing namespace issues

Commit:
implementing d' calculations

Commit:
moving SAAT condition label switching into load_ace rather than proc_ace

Commit:
deleting old versions of functions

Commit:
respelling w/ quasiquotation; completing missing instances of crossed conditions

Commit:
creating super_spread

Commit:
adding some rlang to the namespace

Commit:
now throws warnings for modules that don't process but passes others through

Commit:
fixing roxygen syntax

Commit:
adding todos, fixing capitalization assumption typos

Commit:
adding templates for some SEA task coding

Commit:
create file, for helpers in SEA loading

Commit:
adding summary function to calculate SEA reading score

Commit:
standardized_proc_column_names (to be deprecated) now returns names

Commit:
to calculate SEA reading score (n right - n wrong)

Commit:
whoops! adding todo

Commit:
create ALL_POSSIBLE_DEMOS instance var

Commit:
create master function to compute new trialwise info

Commit:
all modules now functional

Commit:
tidyverse-ing

Commit:
tidyverse-ing; first functional build

Commit:
fixing importFrom

Commit:
adding many dependencies

Commit:
updating documentation

Commit:
3.0.0 for SEA functionality! updating dependencies

Commit:
fixing nest/unnest (from tidyr, not purrr)

Commit:
importing nest/unnest from tidyr, not purrr

Commit:
supposed to help with testing environment issues?

Commit:
adding trycatch to use rbind.fill if bind_rows fails

Commit:
fix bug where data w/o funky delimiters doesn't get re-headered

Commit:
updating missing dependency

Commit:
general debug; actually handles when modules fail

Commit:
again fixing dependencies

Commit:
wide demos strips "file" before joining (such that one subject's data can return in one row)

Commit:
fixing cost-calc-ing modules to not duplicate .overall cols

Commit:
now pipe-safe (1st arg df & returns df)

Commit:
filter_vec & filter_out_vec now accept char vector for pattern

Commit:
vectorizing flatten_df_list

Commit:
vectorizing identify_module

Commit:
first push to add explicit acc/etc recoding of raw data

Commit:
tidy-fied & re-integrated proc_by_module so it now runs both ACE and SEA

Commit:
updating man

Commit:
VECTORIZED :)

Commit:
debugging tidy updates

Commit:
slightly tidying internals

Commit:
now outputs tibble with list-col for data by module

Commit:
implementing standardize_names() with pipe

Commit:
remove_empty_cols now tidy; removes NA and blank cols

Commit:
fixing man links

Commit:
moving sample_data call inside test function

Commit:
V3.1.0: raw data now outputs as list-col separated by module, plus many internal changes

Commit:
adding boxr to ignores in case we can add functionality later

Commit:
redoing authors & urls

Commit:
more import dependencies

Commit:
bugfixes + "data" col now has names attr

Commit:
now creates previous_correct_button

Commit:
filter now outputs fully spread by targs x dists

Commit:
changing rm_short_subs default to false atm (need to debug sea rm short subs)

Commit:
fixing namespace issue (can't importFrom base)

Commit:
previous_correct_button calculated for sea now

Commit:
using str_replace instead of str_remove for looser version reqs on stringr?

Commit:
should cease to throw warning if called on already numeric input

Commit:
disabling quoting within cell, some SEA files don't like it

Commit:
slowly eradicating unnecessary for loops

Commit:
hard-code scrub span subjs with span < 3

Commit:
now REALLY doesn't throw warnings; dplyr::recode is stricter about typing

Commit:
add escape for rm_short_subs in sea running memory span (doesn't return a .overall condition)

Commit:
read_sea_csv patching for reading comp now more robust to varying ncol of "true" data

Commit:
should throw warning instead of error when some files fail to load

Commit:
namespace dependency fix

Commit:
fixing syntax derps for possibly

Commit:
fixing namespace dependency issue

Commit:
slightly rewriting the possibly call

Commit:
fixing tibbley/namespace issues with module_brt/stroop

Commit:
adding sym versions of instance vars, beginning of full qq overhaul

Commit:
beginning data.table implementation removal

Commit:
updating R dependency to require at least 3.4.0

Commit:
minor fix (robust to str-comparing datetime objs)

Commit:
creating instance var for bid_short

Commit:
now creates bid_short

Commit:
fixing issue where to_numeric would fail on actual numbers coded as char

Commit:
shortening code in get_valid_demos()

Commit:
using bind_cols to bind costs; tnt now left_joins to avoid duplicating bid column

Commit:
fixing disambiguation of bid in ACE output wide

Commit:
using rlang::set_names instead of my own one

Commit:
man updates since symbol instance vars added

Commit:
export_csv now tidy

Commit:
improving functionality for converting acc and rt cols

Commit:
adding imports

Commit:
now implements quasiquotation fully

Commit:
now implements quasiquotation fully, scrubs some extraneous conditions before return

Commit:
now pass symbols into the summarizer functions

Commit:
adding "group" (counterbalancing order) label to SEA demos

Commit:
adding support for separate all possible ace and sea demos

Commit:
deprecating parse_subsections_pulvinar; tidy-ing stuff

Commit:
changing from [, COL] to [[COL]] indexing

Commit:
data.table no longer imported! (still required for fread though)

Commit:
Versioning up: Pulvinar now tidy-implemented, and does NOT clean "bad data". Many functions tidier under the hood

Commit:
cleaning syntax in transform_raw

Commit:
no longer fails on records with empty cond blocks

Commit:
adding bid_short & block_half for pulvinar

Commit:
moving half to do by module/bid, not just by bid

Commit:
roxygen2 updates (UTF-8 encoding)

Commit:
updating notes to self

Commit:
now relabels SEA "No Answer"s as "no_response"

Commit:
adding NR detection to correct_button, plus rlang-ing

Commit:
making instance var for short RT cutoff, changing to 150

Commit:
fixing filter so 180 cue rotation is not rotated, coding correct as character earlier

Commit:
actually relabeling correct_button for 180 degree "change"

Commit:
adding post processing helper functions

Commit:
removing rm_short_subs from here

Commit:
adding demographics as module (for ace explorer)

Commit:
repair post_clean_chance, begin post_clean_low_trials

Commit:
updating man

Commit:
removing redundant nest(-module) call

Commit:
again, removing accidental redundant map call

Commit:
first pass version of bad subjs ext scrub

Commit:
helper functions update

Commit:
adding col_time catch for ace explore?

Commit:
usual namespace update

Commit:
updating load_ace for ace explore separate demos

Commit:
partially handles ace explore separate demos

Commit:
handles new pulvinar sea (has diff formatting)

Commit:
one weird trick to make this map work

Commit:
now properly nests by module ACROSS file

Commit:
ace_count now throws out "no_response"

Commit:
namespace update-age

Commit:
no longer tries to repair filter when degree_of_change absent

jessicayounger committed:
Merge pull request #1 from joaquinanguera/master

Getting up to date with master

jessicayounger committed:
Move no response RT code

Moved the code to recode RTs that are evenly divisible by 10 to no response higher in the script. 
Having this last affected coding of trial_accuracy in that for trials with RTs evenly divisible by 10, they would be marked as 'no_response' in correct_button, but not as 'miss' in trial accuracy

monicathieu committed:
Merge pull request #3 from jessicayounger/master

Move no response RT recode

Commit:
updating to import tibble

Commit:
binds ace explore demos to proc using pid

Commit:
deprecating rm_outlier_rts params (now in later f)

Commit:
Merge branch 'master' of https://github.com/joaquinanguera/aceR

jessicayounger committed:
RearrangeRecoding

Moved around correct button and late response coding to get dprime to output correctly. Particularly a problem with T1 where no responses had max rt window put in for their RT instead of NA

jessicayounger committed:
Merge pull request #4 from jessicayounger/master

RearrangeRecoding

Commit:
partial fix of non-numeric correct_button biz

Commit:
Merge branch 'master' of https://github.com/joaquinanguera/aceR

Commit:
reorder bid_short, force lowercase pid

Commit:
try two different datetime orders

Commit:
updating namespace per usual

e-leib committed:
Update load-sea-constants-modules.R

For FRAC_3 module, one of the features of the problems was named (and therefore coded) incorrectly, though the logic was correct. Problems can be congruent or incongruent. A problem is congruent when the larger fraction also has the larger components (e.g., 3/7 > 2/6). A problem is incongruent when the larger fraction has smaller components (e.g., 3/4 > 4/7). In the old code, this was incorrectly named num_larger.

e-leib committed:
Update module-modules-sea.R

Updated the variable name references to reflect new column name and levels in load-sea-constants-modules.R. Now congruency column (formerly known as num_larger).

monicathieu committed:
Merge pull request #6 from e-leib/patch-1

Update module-modules-sea.R to recode fractions level 3 condition from `num_larger` to `congruency`

monicathieu committed:
Merge pull request #5 from e-leib/patch-1

Update load-sea-constants-modules.R to recode fractions level 3 condition from `num_larger` to `congruency`

Commit:
now works with cols with some NAs

Commit:
ace_count should work with non-character cols

Commit:
proc_by_condition excludes redundant acc metrics

Commit:
groupitizing update in progress

Commit:
Merge branch 'master' of https://github.com/joaquinanguera/aceR

Commit:
ace explore carries bid through but joins by pid

Commit:
binds all explore demos; fixes explore brt hiccup

Commit:
now distincts by trial onset time also

Commit:
patch arithm ver, now processes subjs w/o both blocks

jessicayounger committed:
fix cost so always hard - easy

Rearranged condition names so that costs would always have hard condition - easy condition (most were coded as easy - hard).

jessicayounger committed:
Fixing hard costs

Skipped math fluency when correcting so that all costs are hard - easy. Now all costs are hard - easy

monicathieu committed:
Merge pull request #7 from jessicayounger/master

Updating cost functions to reflect hard - easy

Commit:
Light updates. Anticipating classroom branch!

Commit:
Merge branch 'master' of https://github.com/joaquinanguera/aceR
